TLM/TLM/Manager/Impl/RoutingManager.cs

```diff
@@ -1122,6 +1122,21 @@ public class RoutingManager : AbstractGeometryObservingManager, IRoutingManager
 								if (debugFine)
 									Log._Debug($"RoutingManager.RecalculateLaneEndRoutingData({segmentId}, {laneIndex}, {laneId}, {startNode}): -> compatibleLaneDist={compatibleLaneDist}");
 #endif
+                                //disable lane change at the edge of the map.
+                                if (Options.disableLaneChangeAtEdge)
+                                {
+                                    bool isEdgeOfTheMap = false;
+                                    isEdgeOfTheMap |= Math.Abs(Singleton<NetManager>.instance.m_segments.m_buffer[nextSegmentId].m_middlePosition.x) > 8000;
+                                    isEdgeOfTheMap |= Math.Abs(Singleton<NetManager>.instance.m_segments.m_buffer[nextSegmentId].m_middlePosition.z) > 8000;
```

Comment:
Some mods to check for outside-connection related stuff:

* [Unlimited Outside Connections](https://steamcommunity.com/linkfilter/?url=https://github.com/bloodypenguin/Skylines-UnlimitedOutsideConnections)
* [Any road outside connection](https://steamcommunity.com/linkfilter/?url=https://github.com/bloodypenguin/Skylines-UnlimitedOutsideConnections)
* [Cross the line](https://steamcommunity.com/sharedfiles/filedetails/?id=498386331&searchtext=cross+the+line)

Reply:
What about checking if segment has Outside flag?

Reply:
Performance should not be an issue since lane transitions are recalculated on every network change. But the code should be moved inside the `if ((nextLaneInfo.m_finalDirection & nextDir2) != NetInfo.Direction.None) { ... }` block I think. And please do not use `|=` because it has to evaluate both statements every time ;-)